portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,18 +31,68 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  List<HashMap<String, String>> comments = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<String> comments = new ArrayList<>();
-    comments.add("Hey !!! Just commenting - Prakash");
-    comments.add("Roses are red - Nirmala");
-    comments.add("Violets are blue - Sushant");
-    comments.add("What I am doing, I have no clue- Prashant");  
-    
+    response.setContentType("application/json");
+
     Gson gson = new Gson();
     String json = gson.toJson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String firstName = getParameter(request, "first-name", "");
+    String lastName = getParameter(request, "last-name", "");
+    String commentText = getParameter(request, "comment", "");
+    Date commentDate = new Date();
+
+    Entity commentEntity = new Entity("comment");
+    commentEntity.setProperty("firstName", firstName);
+    commentEntity.setProperty("lastName", lastName);
+    commentEntity.setProperty("commentText", commentText);
+    commentEntity.setProperty("date", commentDate);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    HashMap<String, String> commentsData = makeHashmapOfFields(firstName, lastName, commentText);
+
+    comments.add(commentsData);
+    
+    response.sendRedirect("/index.html#connect");
+  }
+
+/**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
```

Comment:
nit: I think the javadoc format is a bit off here and below, could we format/align?